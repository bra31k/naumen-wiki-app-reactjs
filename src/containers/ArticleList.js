import React, { Component } from 'react';
import { connect } from 'react-redux'

import Article from '../components/Article'


class ArticleList extends Component {

    render() {
        return (
            <div>
                { this.props.articles.isLoading
                    ? <p>Загружаю....</p>
                    : this.props.articles.error.length === 0 ? this.props.articles.articles.map(article => {
                        return (
                            <Article
                                title={article.title}
                                snippet={article.snippet}
                                key={article.pageid}
                            />
                        )})
                        : <p>Произошла ошибка: {this.props.articles.error}</p>}
            </div>
        );
    }
}

const mapStateToProps = store => {
    return {
        articles: store,
    }
}

export default connect(
    mapStateToProps
)(ArticleList)
