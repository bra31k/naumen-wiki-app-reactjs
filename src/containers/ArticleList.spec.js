import React from 'react'
import { ArticleList } from './ArticleList'
import { shallow } from 'enzyme'

describe('ArticleList', () => {
    const props = {
        articles: {
            isLoading: false,
            error: null,
            articles: [],
        }
    }

    describe('ArticleList initial', () => {
        const articleList = shallow(<ArticleList {...props} />)

        it('renders properly', () => {
            expect(articleList).toMatchSnapshot()
        })
    })


    describe('ArticleList isLoading', () => {

        let nextProps = {
            articles: {
                isLoading: true,
                error: null,
                articles: [
                    {title: 'hello', snippet: 'world'}
                ],
            }
        }

        const articleList = shallow(<ArticleList {...nextProps} />)

        it('renders properly', () => {
            expect(articleList).toMatchSnapshot()
        })

        it('renders preloader', () => {
            expect(articleList.find('p').first().text()).toEqual('Загружаю....')
        })
    })

    describe('ArticleList error', () => {

        let nextProps = {
            articles: {
                isLoading: false,
                error: 'error',
                articles: [
                    {title: 'hello', snippet: 'world'}
                ],
            }
        }

        const articleList = shallow(<ArticleList {...nextProps} />)

        it('renders properly', () => {
            expect(articleList).toMatchSnapshot()
        })

        it('renders error', () => {
            expect(articleList.find('p').first().text()).toEqual(`Произошла ошибка: ${nextProps.articles.error}`)
        })
    })

    describe('ArticleList render Article', () => {

        let nextProps = {
            articles: {
                isLoading: false,
                error: null,
                articles: [
                    {title: 'hello', snippet: 'world'}
                ],
            }
        }

        const articleList = shallow(<ArticleList {...nextProps} />)

        it('renders properly', () => {
            expect(articleList).toMatchSnapshot()
        })

        it('renders Article', () => {
            expect(articleList.find('Article')).toHaveLength(nextProps.articles.articles.length)
        })
    })

})