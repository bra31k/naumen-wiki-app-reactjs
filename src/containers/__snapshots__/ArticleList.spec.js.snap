// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList ArticleList error renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "articles": Array [
          Object {
            "snippet": "world",
            "title": "hello",
          },
        ],
        "error": "error",
        "isLoading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Произошла ошибка: 
        error
      </p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Произошла ошибка: ",
          "error",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Произошла ошибка: ",
        "error",
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Произошла ошибка: 
          error
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Произошла ошибка: ",
            "error",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Произошла ошибка: ",
          "error",
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ArticleList ArticleList initial renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "articles": Array [],
        "error": null,
        "isLoading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ArticleList ArticleList isLoading renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "articles": Array [
          Object {
            "snippet": "world",
            "title": "hello",
          },
        ],
        "error": null,
        "isLoading": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Загружаю....
      </p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Загружаю....",
      },
      "ref": null,
      "rendered": "Загружаю....",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Загружаю....
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Загружаю....",
        },
        "ref": null,
        "rendered": "Загружаю....",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ArticleList ArticleList render Article renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "articles": Array [
          Object {
            "snippet": "world",
            "title": "hello",
          },
        ],
        "error": null,
        "isLoading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Article
          snippet="world"
          title="hello"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "snippet": "world",
          "title": "hello",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Article
            snippet="world"
            title="hello"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "snippet": "world",
            "title": "hello",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
